### ISSUE TITLE
`editor.autoClosingTags` does not exist in VS code. "Supported VS Code settings" needs to be corrected.

### ISSUE THREAD

**Issue by ion-mironov** [redhat-developer/vscode-xml#961]
Hello,

I'm trying to get the XML validation to work with this extension but it is not doing anything, despite trying to follow the instructions.

The **Supported VS Code settings** part of the extension's description says to disable `editor.autoClosingTags`; that parameter does not exist in the latest VS Code. There is _Brackets_, _Comments_, _Delete_, _Overtype_, and _Quotes_. I've also included a screenshot of all options. Even manually adding in `editor.autoClosingTags` in my settings file has it become dim, signaling that no such option exists.

![autoclose options](https://github.com/redhat-developer/vscode-xml/assets/97184530/3aec90fa-1fea-4bbe-8de4-34cdc0323d73)

I recommend that the instructions be corrected. Also, there is a spelling issue with it; it says "turned _of_ for this" when it should say "turned **off** for this".

**Comment by angelozerr** [comment]
> I'm trying to get the XML validation to work with this extension but it is not doing anything, despite trying to follow the instructions.

Could you share your XML please?

Are you using Java or binary mode?

> that parameter does not exist in the latest VS Code

Indeed but it should not break validation or other features.

**Comment by ion-mironov** [comment]
I _think_ I'm running in binary mode? I actually searched for that online on how to tell if I was or not, but I do have the [Extension Pack for Java](https://marketplace.visualstudio.com/items?itemName=vscjava.vscode-java-pack) extension.

The XML files I'm working with are probably not what most people are used to when someone mentions 'XML'. These particular XML files are for the game SnowRunner, developed by Saber Interactive. I've attached a  ZIP file with an example XML file that is for one of the trucks that is a mod for the game.

Basically, what I was trying to achieve with the extension in VS Code is this: _Visual Studio 2022_ uses whatever default schema or IntelliSense to notice if there is a line/block of code that was not closed properly. For example, in the below picture I removed the closing forward slash to show you what I mean. Notice how it gives the error squiggles and respective explanations for the parts affected, along with marking it with a red square in the Map Mode window to the right.

(**NOTE:** Ignore the error squiggle for the `<Truck>` element. This is **normal** for all the game files and it does not affect anything.)
![first_error_example](https://github.com/redhat-developer/vscode-xml/assets/97184530/b1470057-6072-4ab0-aa2b-5f58766aad79)
![second_error_example](https://github.com/redhat-developer/vscode-xml/assets/97184530/0a68645d-a64d-4609-a2af-248a65657399)


When I put the closing slash back in place, the errors disappear.
![no_error_example](https://github.com/redhat-developer/vscode-xml/assets/97184530/b0d2decf-7abb-48f2-9c7e-cdb49d05ba6c)

This helps me immensely when trying to find out why a mod isn't working; usually it's always due to an end tag missing. I'm wanting the same effect with _**VS Code**_, and was hoping this extension would do that. However, it doesn't and I cannot figure out why; maybe I'm missing something. I've looked over the extension's documentation for a few hours yesterday, along with searching online, and could not find any helpful solution or suggestion. I don't know if it has something to do with incorporating XML schema files or something else. I see that _Visual Studio 2022_ doesn't show a selected schema in the Properties window for the XML file, and clicking on the ellipse menu displays all XML Schemas.
![schema_list](https://github.com/redhat-developer/vscode-xml/assets/97184530/d47d7cde-d63b-4f96-a6fe-fc3413fe408e)

Last night, I was getting confused with not understanding if I had to manually incorporate the same XML Schemas list with **_VS Code_** as a catalog, or have it point to the same directory _Visual Studio 2022_ uses (C:\Program Files\Microsoft Visual Studio\2022\Community\xml\Schemas), or have the extension settings point to the Target Namespaces, or what.

↓ Attached XML file ↓
[Wolf.zip](https://github.com/redhat-developer/vscode-xml/files/14024343/Wolf.zip)

**Comment by fbricon** [comment]
I see an error when removing the closing slash for <RearWheel>
<img width="1904" alt="Screenshot 2024-01-23 at 15 36 41" src="https://github.com/redhat-developer/vscode-xml/assets/148698/b4768f92-576a-436a-aa9c-e95872e498d9">
 
 
 You can check which version, JVM or Binary, of the server is running in the Output view, select `XML Support`. Search for "LemMinX Server info". If you have Java 8+ installed on your machine, and xml.server.preferBinary set to false (by default), then it'll be launched  with a JDK (see above). If you don't have Java installed or binary is preferred, then you'd see a message like:
 
 > Message: Initializing XML Language server
> LemMinX Server info:
> - Version : 0.27.1-SNAPSHOT
> - Native Image
> - VM Version : 17.0.8
> - Git 6b12cd0 - XML Files support with settings
 
Check for any errors in the Output view

**Comment by ion-mironov** [comment]
This is what I got:

> [Info  - 9:53:59 AM] Jan 23, 2024 09:53:59 org.eclipse.lemminx.XMLLanguageServer initialize()
> Message: Initializing XML Language server
> LemMinX Server info:
>  - Version : 0.26.1
>  - Native Image
>  - VM Version : 17.0.2
>  - Git 8379294 - [maven-release-plugin] prepare release 0.26.1

**Comment by fbricon** [comment]
Indeed, that's a limitation of the language server at the moment. It might be possible to improve it, but we're currently full with other obligations. Feel free to contribute a fix.

Now, does validation work as you'd expect, when opened in a folder?

**Comment by fbricon** [comment]
To be clear, opening a single xml file in VS Code should work without opening a folder. Drag'n drop from an archive probably not.

**Comment by ion-mironov** [comment]
> Now, does validation work as you'd expect, when opened in a folder?

Sorry, I forgot to mention that it does **_not_** work. I should've included that XML Support is not listed as an option in the Output window if I simply open the file; I have to open the folder that contains the file in order for XML Support to show up.

> Indeed, that's a limitation of the language server at the moment.

Does that mean that there is currently no way to get **VS Code** to behave the same way that _Visual Studio 2022_ does with validating XML files (the examples I provided at the beginning) ? I don't care if I have to have a folder that contains the XML files, I'm okay with that. I'm more focused on how _Visual Studio 2022_ accomplishes the validation and wanting VS Code to act the same way. Is that possible?
